core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -135,6 +137,19 @@ public <R> R doWithTopicOps(ThrowableFunction<TopicCatalog, R> fn) throws Except
           });
     }
 
+    public <R> R doWithPartitionOps(
+        NameIdentifier tableIdent, ThrowableFunction<SupportsPartitions, R> fn) throws Exception {
+      return classLoader.withClassLoader(
+          cl -> {
+            Preconditions.checkArgument(
+                asTables() != null, "Catalog does not support table operations");
+            Table table = asTables().loadTable(tableIdent);
```

Comment:
could we remove `loadTable` operation to reduce the IO? seems not necessary to do this at first eyes.

Reply:
Partition operations are based on the table, so loading the table is necessary first. If the table does not exist, partition operations are meaningless.

Reply:
when do partition operations we didn't use anything from table except table name(correct me if miss something), so I think there is no need to load table.

Reply:
It is impossible to do partition operations without loading the table unless we redesign the partition management APIs, but the partition management interface was [designed based on the table](https://docs.google.com/document/d/1bWra-dmHizqncxl0zk82GSgMGXIsgvVqw3_O8Jr8OmA/edit#heading=h.xmx8l02atqml) from the beginning.

Reply:
ok, let's keep the current implement since API change is expensive